### PR TITLE
console: dependencies cleanup

### PR DESCRIPTION
**Pull request by beerose**
### Description

‼️ **This PR upgrades Node from 8.x to 12.x.**

**Updated major versions:**
* `prettier` (formatting was run after upgrading).
* `lint-staged` (removed `"git add"` from lint-staged config in package.json, since from v10 onwards the `git add` part is automatic and not necessary to include in configuration),
* `react-autosuggest`,
* `react-tabs`.

**Dependencies that require additional work:**
* `react-router-redux` — not maintained anymore, replace with `connected-react-router`.
* `react-bootstrap` — lots of breaking changes, needs to be discussed if we want to upgrade. Pros: `react-bootstrap` v1 includes TypeScript typings.
* `react-helmet` — upgrade to the newest version, adjust code to the new API.
* `react-table` — completely different API; we may want to stick to the older version for now.
* `react-tabs` — investigate if can be removed/replaced. Currently, it's only used by two components, but one of them is not rendered (and also can be removed). Also, we have multiple solutions for tabs (react-bootstrap, UIKit).
* `react-redux`, `react-notification-system-redux` — upgrading caused some hot reload issues. 
* `graphql` — upgrade from v14 -> v15 introduces breaking changes. 
* `history` — dependent on `react-router`.
* `react-redux` — newer version introduces new API.
* `react-router` — replace with `react-router-dom`.
* `less-loader` — needs investigation.
* `resolve-url-loader` — needs investigation.
* `sass-loader` — needs investigation.
* `webpack-isomorphic-tools` — deprecated.

**Dependencies removed**:
* `deep-equal`,
* `hoist-something`,
* `invariant`,
* `lru-memoize`,
* `map-props`,
* `multireducer`,
* `match-sorter`,
* `pretty-error`,
* `query-string`,
* `uuid`,
* `valid-url`,
* `react-a11y`,
* `react-addons-test-utils`,
* `sinon`,
* `timekeeper`,
* `unused-files-webpack-plugin`,
* `webpack-bundle-analyzer`.

**Unused babel plugins removed from `.babelrc` and dependencies list**:
* `@babel/plugin-syntax-dynamic-import`,
* `@babel/plugin-syntax-import-meta`,
* `@babel/plugin-proposal-json-strings`,
* `@babel/plugin-proposal-decorators`,
* `@babel/plugin-proposal-function-sent`,
* `@babel/plugin-proposal-export-namespace-from`,
* `@babel/plugin-proposal-numeric-separator`,
* `@babel/plugin-proposal-throw-expressions`,
* `@babel/plugin-proposal-logical-assignment-operators`,
* `@babel/plugin-proposal-optional-chaining`,
* `@babel/plugin-proposal-pipeline-operator`,
* `@babel/plugin-proposal-nullish-coalescing-operator`,
* `@babel/plugin-proposal-do-expressions`,
* `@babel/plugin-proposal-function-bind`.

---

This PR is created on top of https://github.com/hasura/graphql-engine/pull/4677.

### Changelog

- [ ] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Console

### Related Issues

https://github.com/hasura/graphql-engine/issues/4189

### CODE REVIEW

**Comment by beerose** [comment]
I don't think we need to convert the whole codebase. There is nothing wrong
with class component. The only thing we should do is replace unsafe
functions.

**Comment by wawhal** [comment]
@beerose I prefer functional components over class components due to the following:

1. It is easier to read effects atomically rather than scanning the lifecycles figure out what's happening
2. Stateful logic cannot exist out of the component in class components and hence cannot be made reusable. This is a big dealbreaker really.
3. Worrying about `this` is unnecessary
4. Functional components were introduced by the React team to solve problems of class components, which by design, make them the better choice.

Nothing's blocked on it so it's not high priority, but it'll make sense to keep this as a community task so that it keeps happening on the side. I can review these in case there's lack of bandwidth.

**Comment by beerose** [comment]
Of course, functional components are better choice. There's no doubt in
that. What is disagree with is giving this to the community. Refactoring
class components to functional ones is error prone, thus I think it should
be done along with some other work on the particular part of the code. So
I'd rather have a task: "Refactor X" which would be about simplifying the
data flow, refactoring react flow, etc., *and* changing class components to
functional.

File: console/src/components/Services/Data/Function/Modify/ModifyCustomFunction.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ class ModifyCustomFunction extends React.Component {
     ]);
   }
 
-  componentWillReceiveProps(nextProps) {
+  UNSAFE_componentWillReceiveProps(nextProps) {
```
Comment (soorajshankar) [inline console/src/components/Services/Data/Function/Modify/ModifyCustomFunction.js:65]: we should slowly get rid of this.

Comment (beerose) [reply console/src/components/Services/Data/Function/Modify/ModifyCustomFunction.js:65]: That's true

**Pull request merged**